api/include/opentelemetry/common/macros.h

```diff
@@ -99,7 +99,8 @@
 #elif defined(__GNUC__)
 #  define OPENTELEMETRY_API_SINGLETON __attribute__((visibility("default")))
 #elif defined(_MSC_VER)
-#  define OPENTELEMETRY_API_SINGLETON
+/* Tentative fix */
+#  define OPENTELEMETRY_API_SINGLETON __declspec(selectany)
```

Comment:
Just wondering why `__declspec(selectany)` is removed?Doesn't it work with MSVC on Windows?

Reply:
@owent 

`__declspec(selectany)` works when applied to a member.

This PR tested applying it to a method, which failed to compile.

See places where `OPENTELEMETRY_API_SINGLETON` is used in this patch,
which are different compared to PR#1525

Reply:
Could we also add `__declspec(selectany)` to variables, so we can also support dll on Windows with MSVC?